### ISSUE TITLE
Prevent overfitting in linear probing

### ISSUE THREAD

**Issue by kutayeroglu** [kutayeroglu/ijepa#27]
> Added flat dataset structure support and pointed to the ground truth file with following commits: 
> 716db55ea98f3ff2725e7e21ed6877e4ae32972b
> 7b0f914567d6b2ecbee27a7d973b35c9d067cda3
> 
> Linear probing script seems to work, output for first epoch is provided below:
> 
> ```
> pyxis: imported docker image: ghcr.io#kutayeroglu/ijepa
> 2025-10-30 10:15:06,718 [__main__:INFO] Dataset directory: /datasets/imagenet-object-localization-challenge/ILSVRC/Data/CLS-LOC
> 2025-10-30 10:15:06,718 [__main__:INFO] Using data from: /datasets/imagenet-object-localization-challenge/ILSVRC/Data/CLS-LOC
> /users/kutay.eroglu/projects/ijepa/main_linprobe.py:111: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
>   checkpoint = torch.load(model_path, map_location="cpu")
> 2025-10-30 10:15:10,979 [__main__:INFO] Loaded model successfully from: /users/kutay.eroglu/projects/ijepa/pretrained_models/IN1K-vit.h.14-300e.pth.tar
> 2025-10-30 10:15:18,094 [__main__:INFO] Created model with linear head
> 2025-10-30 10:15:20,606 [src.utils.linprobe_trainer:INFO] Starting linear probe training
> 2025-10-30 10:15:20,606 [src.utils.linprobe_trainer:INFO] Hyperparameters: lr=0.1, epochs=3
> Epoch 1/3 [Train]: 100%|██████████| 200/200 [02:26<00:00,  1.37it/s, acc=28.86%, loss=1.2231]
> Epoch 1/3 [Val]: 100%|██████████| 157/157 [01:55<00:00,  1.36it/s, acc=0.02%]
> 2025-10-30 10:19:43,041 [src.utils.linprobe_trainer:INFO] Epoch 1/3 | Train Acc: 28.86% | Val Acc: 0.02% | LR: 0.100000
> 2025-10-30 10:19:43,042 [src.utils.linprobe_trainer:INFO] New best model! (0.02% > 0.00%)
> 
> ```
> 
> **Validation accuracy seems to be too low, might need to look into it.** 

 _Originally posted by @kutayeroglu in [#8](https://github.com/kutayeroglu/ijepa/issues/8#issuecomment-3466428933)

Results of second epoch strengthen my concerns:

```
Epoch 2/3 [Train]: 100%|██████████| 200/200 [02:27<00:00,  1.36it/s, acc=61.67%, loss=0.6642]
Epoch 2/3 [Val]: 100%|██████████| 157/157 [01:55<00:00,  1.36it/s, acc=0.00%]
```
Validation accuracy drops to zero while training accuracy increases dramatically

**Comment by kutayeroglu** [comment]
**Changed train_frac from 0.005 to 0.1**

Even though time limit mentioned in https://github.com/users/kutayeroglu/projects/4?pane=issue&itemId=136621614&issue=kutayeroglu%7Cijepa%7C29 prevented further training, initial accuracy for both splits increased.

```
2025-10-30 16:23:18,228 [src.utils.linprobe_trainer:INFO] Hyperparameters: lr=0.01, epochs=5

Epoch 1/5 [Train]: 100%|██████████| 4003/4003 [48:58<00:00,  1.36it/s, acc=57.45%, loss=0.1778]

Epoch 1/5 [Val]: 100%|██████████| 157/157 [01:55<00:00,  1.36it/s, acc=0.02%]

2025-10-30 17:14:12,875 [src.utils.linprobe_trainer:INFO] Epoch 1/5 | Train Acc: 57.45% | Val Acc: 0.02% | LR: 0.010000

2025-10-30 17:14:12,875 [src.utils.linprobe_trainer:INFO] New best model! (0.02% > 0.00%)
```

**Comment by kutayeroglu** [comment]
Should check newly edited description of https://github.com/users/kutayeroglu/projects/4/views/1?pane=issue&itemId=133117355&issue=kutayeroglu%7Cijepa%7C22 or original study's **A.2** section for more details.
